doc/pitfalls/feature_selection.rst

```diff
@@ -0,0 +1,54 @@
+.. _wrong_feature_selection:
+
+================================
+Feature selection using all data
```

Comment:
It seems that the more general pitfall here is leaking test data. There are various examples of this, including feature selection but also scaling, binning, encoding... Basically using any transformer.

So I think it could be worth making this section about testing data leakage in general, and have a bunch of these example (2 should be enough and we can just mention the rest)?

It would also be nice to have a note about using `fit_transform` https://github.com/scikit-learn/scikit-learn/issues/14081#issuecomment-518183137

Reply:
Yes very good point. I did think it was too specific!